cuizinart/main.py

```diff
@@ -1,17 +1,23 @@
+import smtplib
+import ssl
+import traceback
+from email.message import EmailMessage
+
 import requests
 from flask import Flask
 from flask import request
 from flask_cors import CORS
 from flask import jsonify
-from flask import send_file
 from metadata_schema import *
 from settings import *
 
 CORS(app)
 
 basedir = os.path.abspath(os.path.dirname(__file__))
 
+
 def parse_json(obj):
+    request_id = obj['request_id']
```

Comment:
The whole request ID should be generated (and tracked) by the frontend. It needs to be unique. So maybe a user name (email) and a date combination? In CaSPAr we use username and a running number. The request ID is basically the only identifier that both the frontend and backend know.

Reply:
That is a good point. As a user, I don't really want to come up with a request id every time but sometimes it might be a handy way to distinguish requests. How about keeping it optional but generate a random request id if not supplied?

Reply:
Of course we also need to check that the id is new (once we track requests)

Reply:
We can use user email+ current_time_stamp as request_id.

Reply:
I need the request ID on the SLURM backend to create folders. So would appreciate if I would receive a unique ID and don’t have to generate one myself. :)

Reply:
Yes, sure. That's what I meant.
I just thought if a user fires off 5 requests, they might want to give them meaningful names to know which is which when they finish. So my suggestion is to use `email_timestamp_optionalString`.

Reply:
In CaSPAr the user doesn’t set the request ID. It is all done by the frontend. I think it’s too risky if there’s user interaction required for that

Reply:
I don't see why it's risky? But if you think users can manage to distinguish their requests, I'm fine with simplifying the UI and just creating the request in the Cuizinart backend.